lib/scholar/options.ex

```diff
@@ -100,4 +100,12 @@ defmodule Scholar.Options do
     {:error,
      "expected metric to be a :cosine or tuple {:minkowski, p} where p is a positive number or :infinity, got: #{inspect(metric)}"}
   end
+
+  def beta(beta) do
+    if (is_number(beta) and beta >= 0) or (Nx.is_tensor(beta) and Nx.size(beta) == 1) do
```

Comment:
You want here `beta` to be scalar because later on it is used in condition. A tensor that has `Nx.size` equal to 1 is for example ```Nx.tensor([[[[[[1]]]]]])``` ;)
```suggestion
    if (is_number(beta) and beta >= 0) or (Nx.is_tensor(beta) and Nx.rank(beta) == 0) do
```